File: Java/DesignPatterns/src/br/designpattern/singleton/Singleton.java
package br.designpattern.singleton;

public class Singleton {
	
	private static Singleton singletonInstance;
	
	private Singleton() {
		
	}
	
	public static Singleton getSingletonInstance() {
		if(singletonInstance == null) {
			singletonInstance = new Singleton();
		} 		
		return singletonInstance;
	}
	
	public void print() {
		System.out.println("Única instância");
	}
}

File: Javascript/src/singleton/singleton.js
var Singleton = (function () {
    var instance;
 
    function createInstance() {
        var object = new Object("Instância única");
        return object;
    }
 
    return {
        getInstance: function () {
            if (!instance) {
                instance = createInstance();
            }
            return instance;
        }
    };
})();

module.exports = Singleton;